src/site/content/en/blog/webassembly-threads/index.md

```diff
@@ -0,0 +1,472 @@
+emcc -pthread example.c -o example.js
+```
+
+However, when you try to run it in a browser or Node.js, you'll see a warning and then the program
+will hang:
+
+```text
+Before the thread
+Tried to spawn a new thread, but the thread pool is exhausted.
+This might result in a deadlock unless some threads eventually exit or the code
+explicitly breaks out to the event loop.
+If you want to increase the pool size, use setting `-s PTHREAD_POOL_SIZE=...`.
+If you want to throw an explicit error instead of the risk of deadlocking in those
+cases, use setting `-s PTHREAD_POOL_SIZE_STRICT=2`.
+[…hangs here…]
+```
+
+What happened? The problem is, most of the time-consuming APIs on the Web are asynchronous and rely
+on the event loop to execute. This limitation is an important distinction compared to traditional
+environments, where applications normally run I/O in synchronous, blocking manner. Check out the
+blog post about [Using asynchronous web APIs from WebAssembly](https://web.dev/asyncify/) if you'd
+like to learn more.
+
+Particularly in this case, the code synchronously invokes `pthread_create` to create a background
+thread, and follows up by another synchronous call to `pthread_join` that waits for the background
+thread to finish execution. However, Web Workers, that are used behind the scenes when this code is
+compiled with Emscripten, are asynchronous. So what happens is, `pthread_create` only _schedules_ a
+new Worker thread to be created on the next event loop run, but then `pthread_join` immediately
+blocks the event loop to wait for that Worker, and by doing so prevents it from ever being created.
+It's a classic example of a [deadlock](https://en.wikipedia.org/wiki/Deadlock).
+
+One way to solve this problem is to create a pool of Workers ahead of time, before the program has
+even started, so that by the time `pthread_create` is invoked, it only needs to take a ready-to-use
+Worker from the pool, run the provided callback on that background thread and return the Worker back
+to the pool. All of this can be done synchronously, so there won't be any deadlocks as long as the
+pool is sufficiently large.
+
+This is exactly what Emscripten allows with the [`-s
+PTHREAD_POOL_SIZE=...`](https://emsettings.surma.technology/#PTHREAD_POOL_SIZE) option. It allows to
+specify a number of threads - either a fixed number, or a JavaScript expression like
+[`navigator.hardwareConcurrency`](https://developer.mozilla.org/docs/Web/API/NavigatorConcurrentHardware/hardwareConcurrency)
+to create as many threads as there are cores on the CPU. The latter option is helpful when your code
+can scale to an arbitrary number of threads.
+
+In the example above, there is only one thread being created, so instead of reserving all cores it's
+sufficient to use `-s PTHREAD_POOL_SIZE=1`:
+
+```shell
+emcc -pthread -s PTHREAD_POOL_SIZE=1 example.c -o example.js
+```
+
+{% Aside 'success' %} This time, when you execute it, things work successfully: {% endAside %}
```

Comment:
Could you make this aside more descriptive? Something along the lines "When you execute [action] in a [correct way to do it] a [good thing happens (or a bad thing is avoided)]."

I'd also suggest to use "caution" or "gotcha" formatting around the previous example to contrast them.